Statistics/Distributions/IDistributionExtensions.cs

```diff
@@ -87,11 +87,25 @@ public static IDistribution FromXML(XElement ele)
                 {
                     switch (sa.type.Name){
                         case "double":
-                            double vald =  Convert.ToDouble(ele.Attribute(sa.Name).Value);
-                            pi.SetValue(dist,vald);
+                            double vald = 0.0;
+                            if(ele.Attribute(sa.Name).Value=="Infinity"){
+                                vald = double.PositiveInfinity;
+                            }else if(ele.Attribute(sa.Name).Value=="-Infinity"){
+                                vald = double.NegativeInfinity;
+                            }else{
+                                vald = Convert.ToDouble(ele.Attribute(sa.Name).Value);
+                            }
+                            pi.SetValue(dist, vald);
                             break;
                         case "Double":
-                            double valD = Convert.ToDouble(ele.Attribute(sa.Name).Value);
+                        double valD = 0.0;
+                            if(ele.Attribute(sa.Name).Value=="Infinity"){
+                                valD = double.PositiveInfinity;
+                            }else if(ele.Attribute(sa.Name).Value=="-Infinity"){
+                                valD = double.NegativeInfinity;
```

Comment:
can we please use element instead of ele?

Reply:
yes. feel free to update.

Reply:
I suppose I am just attempting to encourage you to think about other aspects of clean code